tests/factories/brac/reports/brac-assessment-follow-up.js

```diff
@@ -0,0 +1,137 @@
+    const ageInYears = moment().diff(patient.date_of_birth, 'years');
+    return ageInYears;
+  })
+  .attr('patient_age_in_months', ['patient'], (patient) => {
+    const ageInMonths = moment().diff(patient.date_of_birth, 'months');
+    return ageInMonths;
+  })
+  .attr('patient_age_in_days', ['patient'], (patient) => {
+    const ageInDays = moment().diff(patient.date_of_birth, 'days');
+    return ageInDays;
+  })
+  .attr('group_followup_options', () => {
+    const groupFollowupOptions = {
+      follow_up_type: Faker.faker.random.arrayElement(['treat', 'treat_refer', 'refer_only']),
+      follow_up_method: Faker.faker.random.arrayElement(['phone', 'in_person']),
+    };
+    return groupFollowupOptions;
+  })
+  .attr('group_danger_signs', ['group_followup_options'], (groupFollowupOptions) => {
+    if (groupFollowupOptions.follow_up_type !== 'treat') {
+      return null;
+    }
+    const groupDangerSigns = {
+      danger_signs: Faker.faker.helpers.uniqueArray(
+        ['convulsions', 'unable_to_feed', 'vomits_everything', 'very_sleepy', 'chest_indrawing'],
+        Faker.faker.datatype.number({ min: 1, max: 5 }))
+    };
+    return groupDangerSigns;
+  })
+  .attr('group_improved',
+    ['group_followup_options', 'group_danger_signs'],
+    (groupFollowupOptions, groupDangerSigns) => {
+      if (groupFollowupOptions.follow_up_type !== 'treat') {
+        return null;
+      }
+      if (!groupDangerSigns || !groupDangerSigns.danger_signs) {
+        const groupImproved = {
+          g_patient_treatment_outcome: Faker.faker.random.arrayElement(
+            ['cured', 'still_recovering', 'bad_medicine_reaction', 'not_improving', 'died'])
+        };
+        return groupImproved;
+      }
+    })
+  .attr('group_referral_followup', ['group_followup_options'], (groupFollowupOptions) => {
+    if (groupFollowupOptions.follow_up_type !== 'treat_refer' && groupFollowupOptions.follow_up_type !== 'refer_only') {
+      return null;
+    }
+    const groupReferralFollowup = {
+      g_patient_health_facility_visit: Faker.faker.random.arrayElement(['yes', 'no'])
+    };
+    return groupReferralFollowup;
+  })
+  .attr('group_better',
+    ['group_followup_options', 'group_referral_followup'],
+    (groupFollowupOptions, groupReferralFollowup) => {
+      if (groupFollowupOptions.follow_up_type !== 'treat_refer'
+        && groupFollowupOptions.follow_up_type !== 'refer_only') {
+        return null;
+      }
+      if (groupReferralFollowup.g_patient_health_facility_visit === 'yes') {
+        return null;
+      }
+      const groupBetter = {
+        g_patient_better: Faker.faker.random.arrayElement(['cured', 'still_recovering', 'still_in_facility', 'died'])
+      };
+      return groupBetter;
+    })
+  .attr('referral_follow_up_needed',
+    ['group_better', 'group_improved', 'group_danger_signs'],
+    (groupBetter, groupImproved, groupDangerSigns) => {
+      if ((groupBetter && groupBetter.g_patient_better === 'no') ||
```

Comment:
This is the value on calculation field in assessment_follow_up.xlsx for **referral_follow_up_needed**:
`if (${g_patient_better}='no' or ${g_patient_treatment_outcome}='bad_medicine_reaction' or ${g_patient_treatment_outcome}='not_improving' or count-selected(${g_danger_signs}) > 0, 'true', 'false')`

And the **type** of **g_patient_better** is **select_one patient_referral_outcome**, with the following values:
- cured
- still_recovering
- still_in_facility
- died

Can this be a bug in the form? I think the field to be used in this calculation instead of **g_patient_better** would be **patient_improved**.

What do you think? But this is not going to correspond to the form calculations.

Reply:
> Can this be a bug in the form

I think it's possible. I think we should signal this to the app services team, or who is managing the brac config. 

In the meantime, I'd rather our code made sense :) 
Either way, we should leave a comment about the discrepancy!